lib/attr_vault/keyring.rb

```diff
@@ -13,15 +17,20 @@ def initialize(id, value)
       end
 
       @id = id
-      @value = value
+      @aead = value.start_with?(AES_256_GCM_PREFIX)
+      @value = if @aead
+                 value[AES_256_GCM_PREFIX.length..-1]
+               else
+                 value
+               end
     end
 
     def digest(data)
       AttrVault::Encryption::hmac_digest(value, data)
     end
 
     def to_json(*args)
-      { id: id, value: value }.to_json
+      { id: id, value: (aead ? AES_256_GCM_PREFIX : '') + value }.to_json
```

Comment:
Although ugly to reconstruct like this, I figure to_json is quite rare, and the rest of the program abstractions hold together well passing around the stripped key text. But, maybe that's too expedient.

Reply:
Nah, that seems legit.